Sources/MongoSwift/ClientSession.swift

```diff
@@ -331,12 +331,18 @@ public final class ClientSession {
             throw ClientSession.SessionInactiveError
         }
 
+        guard let bson = bson_new() else {
+            fatalError("failed to allocate bson_t")
+        }
+        defer { bson_destroy(bson) }
+
         var error = bson_error_t()
-        try doc.withMutableBSONPointer { docPtr in
-            guard mongoc_client_session_append(session, docPtr, &error) else {
-                throw extractMongoError(error: error)
-            }
+        guard mongoc_client_session_append(session, bson, &error) else {
+            throw extractMongoError(error: error)
         }
+
+        let sessionDoc = BSONDocument(copying: bson)
+        doc["lsid"] = sessionDoc["lsid"]
```

Comment:
this is a little hack that I know will work just because I went and looked at the implementation of `mongoc_client_session_append`.

Reply:
When I left this comment, I was actually using the wrong key ("lsid"), it should be "sessionId".